parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1856,19 +1878,29 @@ toSlurpResult uf existingNames =
   -- update (n,r) if (n,r' /= r) exists in names0 and r, r' are Ref
   updates :: SlurpComponent v
   updates = SlurpComponent (Set.fromList types) (Set.fromList terms) where
-    terms = [ var n | (n,r'@Referent.Ref{}) <- R.toList (Names.terms fileNames0)
-                    , [r@Referent.Ref{}] <- [toList $ Names.termsNamed existingNames n]
-                    , r' /= r ]
-    types = [ var n | (n,r') <- R.toList (Names.types fileNames0)
-                    , [r] <- [toList $ Names.typesNamed existingNames n]
-                    , r' /= r ]
+    terms =
+      [ var n
+      | (n, r'@Referent.Ref{}) <- R.toList (Names.terms fileNames0)
+      , [r@Referent.Ref{}]     <- [toList $ Names.termsNamed existingNames n]
+      , r' /= r
+      ]
+    types =
+      [ var n
+      | (n, r') <- R.toList (Names.types fileNames0)
+      , [r]     <- [toList $ Names.typesNamed existingNames n]
+      , r' /= r
+      ]
 
   -- alias (n, r) if (n' /= n, r) exists in names0
   termAliases :: Map v (Set Name)
   termAliases = Map.fromList
     [ (var n, aliases)
     | (n, r@Referent.Ref{}) <- R.toList $ Names.terms fileNames0
-    , aliases <- [Set.delete n $ R.lookupRan r (Names.terms existingNames)]
+    , aliases               <-
+      [ Set.map (Path.unprefixName currentPath) . Set.delete n $ R.lookupRan
```

Comment:
Unprefix the names so they're relative to the current path if possible. This has the effect of deduplicating absolute and relative names.